### DISCUSSION TITLE
Redundant CasperCG commands to alternate PC

### DISCUSSION THREAD

**Discussion by renderfrog** [bitfocus/companion#3176] [Q&A]
How would you go about mirroring a CasperCG connection to an alternative IP. This would serve as a redundant ekstra PC playing out the exact same output as the original PC with the base IP.

Planning to have 2 PCs running identical outputs. And one companion button, when pressed, to simply matrix patch outputs to designated inputs, if something should fail on PC 1.

Torsten@
Videolab

**Comment by dnmeid** [comment]
Set up a second connection for the backup PC. You have to send every action to both connections. If you are using feedbacks you have to decide which feedback you want to use or you can stack them.
There is no built-in support for backup connections that would streamline that workflow because the systems and their needs are just too heterogenous.

**Reply by renderfrog** [reply]
Ok. Thank you.
Would there maybe be a way to have a button change the connection IP address, and still just have one connection to CasparCG configured?
Just a thought.
Maybe a "script" or "function" that changes the conf, or loads a different setup with the backup IP.
It's probably not the case.
But if companion supported an alternate IP for connections. Redundancy would become the real deal for mission critical events.

Torsten
videolab

**Reply by dnmeid** [reply]
There are a lot of possibilities and a lot of thoughts. It is quite a difficult topic. We want to improve redundancy handling somehow but no decision has been made yet.
One of the main problems is, that available actions and options are dynamically generated from the module. Even if I have two times the same device there may be different options available.
